extensions/typescript-language-features/src/languageFeatures/folding.ts

```diff
@@ -35,12 +36,16 @@ class TypeScriptFoldingProvider implements vscode.FoldingRangeProvider {
 			return;
 		}
 
-		return coalesce(response.body.map(span => this.convertOutliningSpan(span, document)));
+		const config = vscode.workspace.getConfiguration(this.modeId);
```

Comment:
In the call to `getConfiguration`, also pass the document as the `scope`. This ensures that the correct configuration is used for the current file